DataFormats/PatCandidates/interface/PATObject.h

```diff
@@ -768,6 +770,24 @@ namespace pat {
   }
 
   template <class ObjectType>
+  void PATObject<ObjectType>::addUserDataObject_( const std::string & label, std::auto_ptr<pat::UserData> & data, bool overwrite ) 
```

Comment:
At a second look, moving to `unique_ptr` would mean either moving `OwnVector` to `auto_ptr` or adding additional methods to `OwnVector` taking `unique_ptr` instead of `auto_ptr` .
@Dr15Jones what do you prefer?